M/Mesa_EGL/build_tarballs.jl

```diff
@@ -0,0 +1,119 @@
+init_block = """
+ENV["LIBGL_DRIVERS_PATH"] = joinpath(artifact_dir, "lib", "dri")
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11"),
+]
+
+# The products that we will ensure are always built
+products = Product[
+    LibraryProduct("libEGL", :libEGL),
+    LibraryProduct("libGLESv1_CM", :libGLESv1_CM),
+    LibraryProduct("libGLESv2", :libGLESv2),
+    LibraryProduct("libGL", :libGL),
+    LibraryProduct("libvulkan_intel", :libvulkan_intel),
+    LibraryProduct("libvulkan_lvp", :libvulkan_lvp),
+    LibraryProduct("libvulkan_radeon", :libvulkan_radeon),
+    LibraryProduct("libxatracker", :libxatracker),
+    LibraryProduct("libgbm", :libgbm),
+    LibraryProduct("libglapi", :libglapi),
+
+    # Drivers
+    LibraryProduct("crocus_dri", :crocus_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("iris_dri", :iris_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("libgallium_dri", :libgallium_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("r300_dri", :r300_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("radeonsi_dri", :radeonsi_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("virtio_gpu_dri", :virtio_gpu_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("i915_dri", :i915_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("kms_swrast_dri", :kms_swrast_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("nouveau_dri", :nouveau_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("r600_dri", :r600_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("swrast_dri", :swrast_dri, ["lib/dri"]; dont_dlopen=true),
+    LibraryProduct("vmwgfx_dri", :vmwgfx_dri, ["lib/dri"]; dont_dlopen=true),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency("Wayland_jll"), # FIXME: HostBuildDependency
+    Dependency("libLLVM_jll"; compat="11.0.0"),
+    HostBuildDependency(PackageSpec(;name="LLVM_jll", version=v"11.0.1")),
+    Dependency("Zlib_jll"),
+    Dependency("libdrm_jll"; compat="2.4.110"),
+    Dependency("Elfutils_jll"),
+    Dependency("Expat_jll"; compat="2.2.10"),
+    Dependency("Zstd_jll"),
+    Dependency("Xorg_libX11_jll"),
+    Dependency("Xorg_xproto_jll"),
+    Dependency("Xorg_kbproto_jll"),
+    Dependency("Xorg_libXext_jll"),
+    Dependency("Xorg_xextproto_jll"),
+    Dependency("Xorg_libXfixes_jll"),
+    Dependency("Xorg_fixesproto_jll"),
+    Dependency("Xorg_libxshmfence_jll"),
+    Dependency("Xorg_glproto_jll"),
+    Dependency("Xorg_dri2proto_jll"),
+    Dependency("Xorg_libXxf86vm_jll"),
+    Dependency("Xorg_xf86vidmodeproto_jll"),
+    Dependency("Xorg_libXrandr_jll"),
+    Dependency("Xorg_randrproto_jll"),
+    Dependency("Xorg_renderproto_jll"),
```

Comment:
The *proto packages are headers-only, so they should be `BuildDependency`.  Also, `Xorg_xorgproto_jll` replaces all of them (we can't remove them from the registry because they were used by old versions of Xorg packages)
```suggestion
    BuildDependency("Xorg_xorgproto_jll"),
    Dependency("Xorg_libX11_jll"),
    Dependency("Xorg_libXext_jll"),
    Dependency("Xorg_libXfixes_jll"),
    Dependency("Xorg_libxshmfence_jll"),
    Dependency("Xorg_libXxf86vm_jll"),
    Dependency("Xorg_xf86vidmodeproto_jll"),
    Dependency("Xorg_libXrandr_jll"),
```